src/Messages/OWSMessageSender.m

```diff
@@ -1089,10 +1102,16 @@ - (void)handleMismatchedDevices:(NSDictionary *)dictionary
 - (void)handleMessageSentLocally:(TSOutgoingMessage *)message
 {
     if (message.shouldSyncTranscript) {
-        // TODO: I suspect we shouldn't optimistically set hasSyncedTranscript.
-        //       We could set this in a success handler for [sendSyncTranscriptForMessage:].
-        [message updateWithHasSyncedTranscript:YES];
-        [self sendSyncTranscriptForMessage:message];
+        __block BOOL hasSecondaryDevices = NO;
+        [self.dbConnection readWithBlock:^(YapDatabaseReadTransaction *_Nonnull transaction) {
+            hasSecondaryDevices = [OWSDevice hasSecondaryDevicesWithTransaction:transaction];
+        }];
```

Comment:
seems like we could pull this into `OWSDevice` like this: `hasSecondaryDevicesWithConnection:`